crates/storage/src/connection/transaction.rs

```diff
@@ -26,23 +26,34 @@ pub(super) fn insert_transactions(
     for (i, (transaction, receipt)) in transaction_data.iter().enumerate() {
         // Serialize and compress transaction data.
         let transaction = dto::Transaction::from(transaction);
-        let receipt = dto::Receipt::from(receipt);
 
         let tx_data = serde_json::to_vec(&transaction).context("Serializing transaction")?;
         let tx_data = compressor
             .compress(&tx_data)
             .context("Compressing transaction")?;
 
-        let serialized_receipt = serde_json::to_vec(&receipt).context("Serializing receipt")?;
-        let serialized_receipt = compressor
-            .compress(&serialized_receipt)
-            .context("Compressing receipt")?;
-
-        let execution_status = match receipt.execution_status {
-            dto::ExecutionStatus::Succeeded => 0,
-            dto::ExecutionStatus::Reverted => 1,
+        let serialized_receipt = match receipt {
+            Some(receipt) => {
+                let receipt = dto::Receipt::from(receipt);
+                let serialized_receipt =
+                    serde_json::to_vec(&receipt).context("Serializing receipt")?;
+                Some(
+                    compressor
+                        .compress(&serialized_receipt)
+                        .context("Compressing receipt")?,
+                )
+            }
+            None => None,
         };
 
+        let execution_status = receipt
+            .as_ref()
+            .map(|receipt| match receipt.execution_status {
+                ExecutionStatus::Succeeded => 0,
+                ExecutionStatus::Reverted { .. } => 1,
+            })
+            .unwrap_or(0);
```

Comment:
Slightly unrelated, but I think we don't actually use this column (in a separate issue/PR). Would be great to remove as storing "succeeded" for transactions without a receipt is not really correct.